ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d_pybind.cpp

```diff
@@ -20,23 +20,24 @@ void bind_max_pool2d_operation(py::module& module) {
-        | kernel_h          | kernel window height          | uint32_t      |             | Yes      |
-        | kernel_w          | kernel window width           | uint32_t      |             | Yes      |
-        | stride_h          | stride in height dim          | uint32_t      |             | No       |
-        | stride_w          | stride in width dim           | uint32_t      |             | No       |
-        | pad_h             | padding in height dim         | uint32_t      |             | No       |
-        | pad_w             | padding in width dim          | uint32_t      |             | No       |
-        | dilation_h        | kernel dilation in height dim | uint32_t      |             | No       |
-        | dilation_w        | kernel dilation in width dim  | uint32_t      |             | No       |
-        | memory_config     | Output memory config          | MemoryConfig  |             | No       |
-        +-------------------+-------------------------------+---------------+-------------+----------+
+        +----------------------+-------------------------------+--------------------+-------------+----------+
+        | Argument             | Description                   | Data type          | Valid range | Required |
+        +======================+===============================+====================+=============+==========+
+        | input                | Input activations tensor      | Tensor             |             | Yes      |
+        | in_n                 | Input nbatch                  | Tensor             |             | Yes      |
+        | in_h                 | Input height                  | Tensor             |             | Yes      |
+        | in_w                 | Input width                   | Tensor             |             | Yes      |
+        | kernel_h             | kernel window height          | uint32_t           |             | Yes      |
+        | kernel_w             | kernel window width           | uint32_t           |             | Yes      |
+        | stride_h             | stride in height dim          | uint32_t           |             | No       |
+        | stride_w             | stride in width dim           | uint32_t           |             | No       |
+        | pad_h                | padding in height dim         | uint32_t           |             | No       |
+        | pad_w                | padding in width dim          | uint32_t           |             | No       |
+        | dilation_h           | kernel dilation in height dim | uint32_t           |             | No       |
+        | dilation_w           | kernel dilation in width dim  | uint32_t           |             | No       |
+        | memory_config        | Output memory config          | MemoryConfig       |             | No       |
+        | applied_shard_scheme | Shard scheme                  | TensorMemoryLayout |             | No       |
```

Comment:
It would be good to add a description (following proper doc layout) of what max pool2d does and which combinations of parameters are valid. E.g. what does applied shard scheme do, when should applied shard scheme be specified.

Also, why is interleaved chosen as the default instead of using std::optional?

Reply:
Borys, I have updated MaxPool2D to use std::optional instead of the default parameter and I think this results in much clearer logic!

I was wondering if you could point me to either the pybind doc layout SOP or a good example I might be able to reference and I can add a description here.

Reply:
You can use ttnn/cpp/ttnn/operations/matmul/matmul_pybind.cpp